### PR TITLE
MRG, BUG: Fix Epochs.reject_tmin, reject_tmax handling upon Epochs creation & cropping

### LINKED ISSUE #8814: Epochs.reject_tmin and reject_tmax aren't being adjusted for deviations due to sampling rate
MWE:

```python
import os
import mne

sample_data_folder = mne.datasets.sample.data_path()
sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
                                    'sample_audvis_raw.fif')

raw = mne.io.read_raw_fif(sample_data_raw_file, verbose=False).crop(tmax=60)
events = mne.find_events(raw, stim_channel='STI 014')

tmin, tmax = -0.2, 0.8
epochs = mne.Epochs(raw, events,
                    tmin=tmin, tmax=tmax,
                    reject_tmin=tmin, reject_tmax=tmax)

print(epochs.tmin)
print(epochs.reject_tmin)
print(epochs.tmax)
print(epochs.reject_tmax)
```

Output:
```python
-0.19979521315838786
-0.2
0.7991808526335514
0.8
```

Saving above Epochs to disk and trying to load them will fail:
```python
epochs_fname = '/tmp/epochs-epo.fif'

epochs.save(epochs_fname)
epochs_loaded = mne.read_epochs(epochs_fname)
```

Output:
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Untitled-1 in 
     18 epochs.save(epochs_fname, overwrite=True)
     19 
---> 20 epochs_read = mne.read_epochs(epochs_fname)
     21 

<decorator-gen-212> in read_epochs(fname, proj, preload, verbose)

~/Development/mne-python/mne/epochs.py in read_epochs(fname, proj, preload, verbose)
   2704         The epochs.
   2705     """
-> 2706     return EpochsFIF(fname, proj, preload, verbose)
   2707 
   2708 

<decorator-gen-213> in __init__(self, fname, proj, preload, verbose)

~/Development/mne-python/mne/epochs.py in __init__(self, fname, proj, preload, verbose)
   2819         # again, ensure we're retaining the baseline period originally loaded
   2820         # from disk without trying to re-apply baseline correction
-> 2821         super(EpochsFIF, self).__init__(
   2822             info, data, events, event_id, tmin, tmax, baseline=None, raw=raw,
   2823             proj=proj, preload_at_end=False, on_missing='ignore',

<decorator-gen-198> in __init__(self, info, data, events, event_id, tmin, tmax, baseline, raw, picks, reject, flat, decim, reject_tmin, reject_tmax, detrend, proj, on_missing, preload_at_end, selection, drop_log, filename, metadata, event_repeated, verbose)

~/Development/mne-python/mne/epochs.py in __init__(***failed resolving arguments***)
    481         # check reject_tmin and reject_tmax
    482         if (reject_tmin is not None) and (reject_tmin < tmin):
--> 483             raise ValueError("reject_tmin needs to be None or >= tmin")
    484         if (reject_tmax is not None) and (reject_tmax > tmax):
    485             raise ValueError("reject_tmax needs to be None or <= tmax")

ValueError: reject_tmin needs to be None or >= tmin
```

### LINKED ISSUE #8816: Epochs.crop() doesn't adjust Epochs.reject_tmin, reject_tmax
Related to #8814

MWE:
```python
import os
import mne

sample_data_folder = mne.datasets.sample.data_path()
sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
                                    'sample_audvis_raw.fif')
epochs_fname = '/tmp/epochs-epo.fif'


raw = mne.io.read_raw_fif(sample_data_raw_file, verbose=False).crop(tmax=60)
events = mne.find_events(raw, stim_channel='STI 014')

tmin, tmax = -0.2, 0.8
epochs = mne.Epochs(raw, events,
                    tmin=tmin, tmax=tmax,
                    reject_tmin=tmin, reject_tmax=tmax)
epochs.load_data()
epochs.crop(-0.1, 0.5)

print(epochs.tmin)
print(epochs.reject_tmin)
print(epochs.tmax)
print(epochs.reject_tmax)
```
Output:
```python
-0.09989760657919393
-0.2
0.49948803289596966
0.8
```

Trying to save & load:
```python
epochs.save(epochs_fname, overwrite=True)
epochs_read = mne.read_epochs(epochs_fname)
```
Output:
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/tmp/test.py in 
     25 
     26 epochs.save(epochs_fname, overwrite=True)
---> 27 epochs_read = mne.read_epochs(epochs_fname)

<decorator-gen-212> in read_epochs(fname, proj, preload, verbose)

~/Development/mne-python/mne/epochs.py in read_epochs(fname, proj, preload, verbose)
   2704         The epochs.
   2705     """
-> 2706     return EpochsFIF(fname, proj, preload, verbose)
   2707 
   2708 

<decorator-gen-213> in __init__(self, fname, proj, preload, verbose)

~/Development/mne-python/mne/epochs.py in __init__(self, fname, proj, preload, verbose)
   2819         # again, ensure we're retaining the baseline period originally loaded
   2820         # from disk without trying to re-apply baseline correction
-> 2821         super(EpochsFIF, self).__init__(
   2822             info, data, events, event_id, tmin, tmax, baseline=None, raw=raw,
   2823             proj=proj, preload_at_end=False, on_missing='ignore',

<decorator-gen-198> in __init__(self, info, data, events, event_id, tmin, tmax, baseline, raw, picks, reject, flat, decim, reject_tmin, reject_tmax, detrend, proj, on_missing, preload_at_end, selection, drop_log, filename, metadata, event_repeated, verbose)

~/Development/mne-python/mne/epochs.py in __init__(***failed resolving arguments***)
    481         # check reject_tmin and reject_tmax
    482         if (reject_tmin is not None) and (reject_tmin < tmin):
--> 483             raise ValueError("reject_tmin needs to be None or >= tmin")
    484         if (reject_tmax is not None) and (reject_tmax > tmax):
    485             raise ValueError("reject_tmax needs to be None or <= tmax")

ValueError: reject_tmin needs to be None or >= tmin
```

### PR DESCRIPTION
**Pull request by hoechenberger**
- Adjust Epochs.reject_tmin, reject_tmax for time inaccuracies due to sfreq. Fixes #8814
- Adjust Epochs.reject_tmin, reject_tmax upon cropping. Fixes #8816

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -526,6 +516,28 @@ def __init__(self, info, data, events, event_id=None, tmin=-0.2, tmax=0.5,
         self._raw_times = np.arange(start_idx,
                                     int(round(tmax * sfreq)) + 1) / sfreq
         self._set_times(self._raw_times)
+
+        # check reject_tmin and reject_tmax
+        if (reject_tmin is not None) and (np.isclose(reject_tmin, tmin)):
+            # potentially adjust for small deviations due to sampling freq
+            reject_tmin = self.tmin
+        elif (reject_tmin is not None) and (reject_tmin < tmin):
+            raise ValueError("reject_tmin needs to be None or >= tmin")
```
Comment (larsoner) [inline mne/epochs.py:525]: Simpler control flow
```suggestion
        if reject_tmin is not None:
            if np.isclose(reject_tmin, tmin):
                # potentially adjust for small deviations due to sampling freq
                reject_tmin = self.tmin
            elif reject_tmin < tmin:
                raise ValueError("reject_tmin needs to be None or >= tmin")
```

File: mne/epochs.py [hunk 2]
```diff
@@ -1560,6 +1572,16 @@ def crop(self, tmin=None, tmax=None, include_tmax=True, verbose=None):
         self._set_times(self.times[tmask])
         self._raw_times = self._raw_times[tmask]
         self._data = self._data[:, :, tmask]
+
+        # Adjust rejection period
+        if self.reject_tmin is not None and self.reject_tmin < self.tmin:
+            warn('reject_tmin is not in epochs time interval. reject_tmin is '
+                 'set to epochs.tmin')
```
Comment (larsoner) [inline mne/epochs.py:1579]: Do we actually need to warn for this? I think `logger.info` is probably more appropriate when cropping.

File: mne/epochs.py (hunk 1 above)
Comment (agramfort) [inline mne/epochs.py:525]:
```suggestion
            raise ValueError(f"reject_tmin needs to be None or >= tmin (got {reject_tmin})")
```

File: mne/epochs.py [hunk 3]
```diff
@@ -526,6 +516,28 @@ def __init__(self, info, data, events, event_id=None, tmin=-0.2, tmax=0.5,
...
+            # potentially adjust for small deviations due to sampling freq
+            reject_tmin = self.tmin
+        elif (reject_tmin is not None) and (reject_tmin < tmin):
+            raise ValueError("reject_tmin needs to be None or >= tmin")
+
+        if (reject_tmax is not None) and (np.isclose(reject_tmax, tmax)):
+            # potentially adjust for small deviations due to sampling freq
+            reject_tmax = self.tmax
+        elif (reject_tmax is not None) and (reject_tmax > tmax):
+            raise ValueError("reject_tmax needs to be None or <= tmax")
```
Comment (agramfort) [inline mne/epochs.py:531]: say what value you got.

File: mne/epochs.py [hunk 4]
```diff
@@ -526,6 +516,28 @@ def __init__(self, info, data, events, event_id=None, tmin=-0.2, tmax=0.5,
...
+
+        if (reject_tmin is not None) and (reject_tmax is not None):
+            if reject_tmin >= reject_tmax:
+                raise ValueError('reject_tmin needs to be < reject_tmax')
```
Comment (agramfort) [inline mne/epochs.py:535]: say what value you got

**Review by agramfort — COMMENTED** [review]
besides LGTM

File: mne/epochs.py (hunk 2 above)
Comment (hoechenberger) [reply mne/epochs.py]: I've changed it to info

File: mne/epochs.py (hunk 1 above)
Comment (hoechenberger) [reply mne/epochs.py]: Addressed in the latest commit

Comment (hoechenberger) [reply mne/epochs.py]: Addressed in the latest commit

**Review by agramfort — APPROVED** [review]
@hoechenberger feel free to MRG

**Pull request merged**